build/downgrade-rector.sh

```diff
@@ -34,7 +34,11 @@ for directory in $directories; do
     echo "[NOTE] Downgrading '$directory' directory\n"
 
     # --working-dir is needed, so "SKIP" parameter is applied in absolute path of nested directory
-    php -d memory_limit=-1 bin/rector process $directory --config build/config/config-downgrade.php --working-dir $BUILD_DIRECTORY --ansi
+    if [[ "$directory" =~ ^(vendor/(symfony|symplify|rector)|src|utils).* ]]; then
+        php -d memory_limit=-1 bin/rector process $directory --config build/config/config-downgrade.php --working-dir $BUILD_DIRECTORY --ansi
+    else
+        php -d memory_limit=-1 bin/rector process $directory --config build/config/config-downgrade-parallel.php --working-dir $BUILD_DIRECTORY --ansi
+    fi
```

Comment:
I think we should keep the former single command, as this will be very confusing for anyone who wants to downgrade.

Reply:
Ok, I will move config to variable.

Reply:
I tried updated script to move to variable:

```bash
    if [[ "$directory" =~ ^(vendor/(symfony|symplify|rector)|src|utils).* ]]; then
        config="build/config/config-downgrade.php"
    else
        config="build/config/config-downgrade-parallel.php"
    fi

    # --working-dir is needed, so "SKIP" parameter is applied in absolute path of nested directory
    php -d memory_limit=-1 bin/rector process $directory --config $config --working-dir $BUILD_DIRECTORY --ansi
```

it seems I got error on php 7.4:

```bash
➜  rector-build git:(downgrade-parallel) ✗ /opt/homebrew/Cellar/php@7.4/7.4.26_2/bin/php bin/rector -vvv --debug

 [ERROR] syntax error, unexpected 'private' (T_PRIVATE), expecting variable (T_VARIABLE) 
```

probably I wrote something wrong in the script that possibly make variable not overriden in next iteration?

Reply:
This is strange, still not work even with unset variable in end of each iteration:

```bash
for directory in $directories; do
    echo "[NOTE] Downgrading '$directory' directory\n"

    if [[ "$directory" =~ ^(vendor/(symfony|symplify|rector)|src|utils).* ]]; then
        CONFIG_PATH_DOWNGRADE="build/config/config-downgrade.php"
    else
        CONFIG_PATH_DOWNGRADE="build/config/config-downgrade-parallel.php"
    fi

    # --working-dir is needed, so "SKIP" parameter is applied in absolute path of nested directory
    php -d memory_limit=-1 bin/rector process $directory --config $CONFIG_PATH_DOWNGRADE --working-dir $BUILD_DIRECTORY --ansi

    unset CONFIG_PATH_DOWNGRADE
done
```